src/pip/_internal/commands/show.py

```diff
@@ -88,7 +95,7 @@ def get_requiring_packages(package_name):
             'required_by': get_requiring_packages(dist.project_name)
         }
         file_list = None
-        metadata = None
+        metadata = ''
```

Comment:
Just to know, what this change would do?

Reply:
This is done to avoid  `metadata` being typed as `Optional[Any]` in function calls using metadata below and causing mypy errors like ` error: Argument 1 to "feed" of "FeedParser" has incompatible type "Optional[Any]"; expected "str"` or `Item "None" of "Optional[Any]" has no attribute "splitlines"`

Also since after this line of code, we have `metada = dist.get_metadata(..)`which returns a string anyways, instantiating metadata as `''` as `None` shouldn't cause any issues.

Reply:
The change from None to '' looks good. I would remove the comment though, as it explain the change but when reading the code it's not that helpful.

Reply:
Agreed that the comment doesn't add much value to what is a trivial change. Have removed it